### PR TITLE
feat: refactor publishTest and archiveTest

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Refactor publishTest and archiveTest](https://www.notion.so/uwblueprintexecs/Refactor-publishTest-and-archiveTest-b7c0726b10634af8b87e979d9f8c4c51?pvs=4)

## Implementation description
* Move logic for publishTest and archiveTest from resolvers to service
* Add new tests and refactor unit tests

## Steps to test
1. yarn test
2. GraphQL

publish
image
image

archive
image
image

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/graphql/resolvers/testResolvers.ts [hunk 1]
```diff
@@ -138,15 +138,7 @@ const testResolvers = {
       _req: undefined,
```
Comment (joyce-shi) [inline backend/graphql/resolvers/testResolvers.ts:137]: could you remove unused import on line 3 please :D

File: backend/services/implementations/testService.ts [hunk 2]
```diff
@@ -147,15 +147,49 @@ class TestService implements ITestService {
 
   async getAllTests(): Promise<TestResponseDTO[]> {
     try {
-      let tests = await MgTest.find();
-      tests = tests.filter((test) => test.status !== AssessmentStatus.DELETED);
+      const tests = await MgTest.find({
+        status: { $ne: AssessmentStatus.DELETED },
+      });
```
Comment (joyce-shi) [inline backend/services/implementations/testService.ts:152]: nice

**Review by joyce-shi — APPROVED** [review]
LGTM! just a quick comment

File: backend/graphql/resolvers/testResolvers.ts (hunk 1 above)
Comment (carissa-tang) [reply backend/graphql/resolvers/testResolvers.ts:137]: oopsies

**Pull request merged**